Source/WebCore/platform/audio/mac/AudioSessionMac.mm

```diff
@@ -483,6 +483,33 @@ static float defaultDeviceTransportIsBluetooth()
     }
 }
 
+size_t AudioSessionMac::outputLatency() const
```

Comment:
I would tend to move this computation to AudioOuptutUnitAdaptor.
The reason is that AudioContext can be set to a particular device via setSinkId.

To handle the case of audio context playing on the device device and default device is changing, AudioOuptutUnitAdaptor would ideally be notified that the default device is changing and would do the recomputation if it is not set to a particular device via setSinkId.
And then send the updated value to its AudioOuptutUnitAdaptor client.

Reply:
We don't currently support setSinkId on the AudioContext so I will take this as a follow-up.

Reply:
I also don't see how we can retrieve the device's latency from the AudioUnit